### PR TITLE
[GRAPE-117] added new flags to launcher.py for select_bad_samples.py config

### PR DESCRIPTION
**Pull request by Jahysama**

### CODE REVIEW

File: scripts/select_bad_samples.py [hunk 1]
```diff
@@ -30,19 +33,18 @@
...
+    psc.loc[:, 'alt_hom_share'] = psc.nNonRefHom / psc.nNonMissing
+    psc.loc[:, 'het_samples_share'] = psc.nHets / psc.nNonMissing
 
-    bad_missing_samples_mask = (psc.missing_share > 0.15) | (psc.nMissing + psc.nNonMissing == 0)
-
-    bad_alt_hom_samples_mask = ((psc.nNonRefHom / psc.nNonMissing) <= 0.01) | (psc.nNonMissing == 0)
+    bad_missing_samples_mask = (psc.missing_share >= missing_samples/100) | (psc.nMissing + psc.nNonMissing == 0)
 
-    bad_het_samples_mask = ((psc.nHets / psc.nNonMissing) <= 0.05) | (psc.nNonMissing == 0)
+    bad_alt_hom_samples_mask = (psc.alt_hom_share <= alt_hom_samples/100) | (psc.nNonMissing == 0)
```
Comment (josephkott) [inline scripts/select_bad_samples.py:41]:
```suggestion
    bad_alt_hom_samples_mask = (psc.alt_hom_share <= alt_hom_samples / 100) | (psc.nNonMissing == 0)
```

File: scripts/select_bad_samples.py [hunk 2]
```diff
@@ -30,19 +33,18 @@
 
     psc.loc[:, 'nNonMissing'] = psc.nRefHom + psc.nNonRefHom + psc.nHets
     psc.loc[:, 'missing_share'] = psc.nMissing / (psc.nMissing + psc.nNonMissing)
+    psc.loc[:, 'alt_hom_share'] = psc.nNonRefHom / psc.nNonMissing
+    psc.loc[:, 'het_samples_share'] = psc.nHets / psc.nNonMissing
 
-    bad_missing_samples_mask = (psc.missing_share > 0.15) | (psc.nMissing + psc.nNonMissing == 0)
-
-    bad_alt_hom_samples_mask = ((psc.nNonRefHom / psc.nNonMissing) <= 0.01) | (psc.nNonMissing == 0)
+    bad_missing_samples_mask = (psc.missing_share >= missing_samples/100) | (psc.nMissing + psc.nNonMissing == 0)
```
Comment (josephkott) [inline scripts/select_bad_samples.py:39]:
```suggestion
    bad_missing_samples_mask = (psc.missing_share >= missing_samples / 100) | (psc.nMissing + psc.nNonMissing == 0)
```

File: launcher.py [hunk 3]
```diff
@@ -411,7 +436,8 @@ def copy_input(input_dir, working_dir, samples_file):
             until=[args.until] if args.until is not None else [],
             use_conda=True,
             conda_prefix=args.conda_prefix,
-            envvars=['CONDA_ENVS_PATH', 'CONDA_PKGS_DIRS']
+            envvars=['CONDA_ENVS_PATH', 'CONDA_PKGS_DIRS'],
+            keepgoing=True
```
Comment (josephkott) [inline launcher.py:440]: How does this relate to the current PR?

File: launcher.py [hunk 4]
```diff
@@ -240,6 +240,24 @@ def get_parser_args():
...
+        "--missing-samples",
+        default=15.0,
+        type=float,
+        help="Percentage of missing SNPs in a sample above or equal which it is filtered out.")
+
+    parser.add_argument(
+        "--alt-hom-samples",
+        default=1.0,
+        type=float,
+        help="Percentage of homozygous alternative SNPs in a sample below or equal which it is filtered out.")
```
Comment (josephkott) [inline launcher.py:253]:
```suggestion
        help='Lower bound of homozygous alternative SNPs (%). Samples with lower values are removed from the relatedness detection analysis.')
```

File: launcher.py [hunk 5]
```diff
@@ -240,6 +240,24 @@ def get_parser_args():
...
+
+    parser.add_argument(
+        "--het-samples",
```
Comment (josephkott) [inline launcher.py:256]:
```suggestion
        '--het-samples',
```

File: scripts/select_bad_samples.py [hunk 6]
```diff
@@ -30,19 +33,18 @@
...
 
-    bad_het_samples_mask = ((psc.nHets / psc.nNonMissing) <= 0.05) | (psc.nNonMissing == 0)
+    bad_alt_hom_samples_mask = (psc.alt_hom_share <= alt_hom_samples/100) | (psc.nNonMissing == 0)
 
-    psc.loc[:, 'alt_hom_share'] = psc.nNonRefHom / psc.nNonMissing
-    psc.loc[:, 'het_samples_share'] = psc.nHets / psc.nNonMissing
+    bad_het_samples_mask = (psc.het_samples_share <= het_samples/100) | (psc.nNonMissing == 0)
```
Comment (josephkott) [inline scripts/select_bad_samples.py:43]:
```suggestion
    bad_het_samples_mask = (psc.het_samples_share <= het_samples / 100) | (psc.nNonMissing == 0)
```

File: launcher.py [hunk 7]
```diff
@@ -240,6 +240,24 @@ def get_parser_args():
...
+        default=5.0,
+        type=float,
+        help="Percentage of heterozygous SNPs in a sample below or equal which it is filtered out.")
```
Comment (josephkott) [inline launcher.py:259]:
```suggestion
        help='Lower bound of heterozygous SNPs (%). Samples with lower values are removed from the relatedness detection analysis.')
```

**Review by josephkott — CHANGES_REQUESTED** [review]

File: launcher.py (hunk 3 above)
Comment (Jahysama) [reply launcher.py:440]: It does not actually, it is related to the GRAPE-118. Pipeline needs to keep going after some batches fall.

Comment (josephkott) [reply launcher.py:440]: Pls move this change to `GRAPE-118`.

Comment (Jahysama) [reply launcher.py]: Ok, done.

**Review by josephkott — APPROVED** [review]

**Pull request merged**